cdap-storage-ext-spanner/src/main/java/io/cdap/cdap/storage/spanner/SpannerStructuredRow.java

```diff
@@ -63,7 +72,20 @@ public Boolean getBoolean(String fieldName) throws InvalidFieldException {
   @Nullable
   @Override
   public String getString(String fieldName) throws InvalidFieldException {
-    return isNull(fieldName) ? null : struct.getString(fieldName);
+    String value = null;
+    if (!isNull(fieldName)) {
+      value = struct.getString(fieldName);
+      String compressor = schema.getCompressor(fieldName);
+      if (value != null && compressor != null) {
+        if (compressor.equals(SNAPPY_COMPRESSOR)) {
```

Comment:
Use a Strategy pattern to perform compression / decompression based on compressor type.

Reply:
Suggesting a pseudo code:

```
enum CompressorType {
    SNAPPY("snappy");
}


// Strategy Interface
interface Compressor {
    String compress(String data) throws IOException;

    String decompress(String compressedData) throws IOException;

   // Similarly, for byte[], if needed
   byte[] compress(byte[] data) throws IOException;

    byte[] decompress(byte[] compressedData) throws IOException;
}

class SpannyCompressor implements Compressor {
    ...
    ...
}

class CompressorFactory {
    static Compressor createCompressor(CompressionType compressionType) {
        switch (compressionType) {
              return new SpannyCompressor();
        }
    }
}

```